64.CppCompat/main.cpp

```diff
@@ -110,7 +163,86 @@ int main()
     float32_t3x3 z_inv = inverse(z);
     auto mid = lerp(x,x,0.5f);
     auto w = transpose(y);
+    
+
+    // half test
+    {
+
+        float16_t MIN = 6.103515e-05F;
+        float16_t MAX = 65504.0F;
+        float16_t DENORM_MIN = 5.96046448e-08F;
+        uint16_t  QUIET_NAN = 0x7FFF;
+        uint16_t  SIGNALING_NAN = 0x7DFF;
+
+        assert(equal(std::numeric_limits<float16_t>::min(), MIN));
+        assert(equal(std::numeric_limits<float16_t>::max(), MAX));
+        assert(equal(std::numeric_limits<float16_t>::denorm_min(), DENORM_MIN));
+        assert(equal(std::numeric_limits<float16_t>::quiet_NaN(), QUIET_NAN));
+        assert(equal(std::numeric_limits<float16_t>::signaling_NaN(), SIGNALING_NAN));
+    }
+
+    auto test_type_limits = []<class T>() 
+    {
+        using L = std::numeric_limits<T>;
+        using R = nbl::hlsl::numeric_limits<T>;
+
+        
+        #define TEST_AND_LOG(var) \
+            { \
+                auto rhs = limits_var(R::var); \
+                auto lhs = limits_var(L::var); \
+                if(!equal(lhs, rhs)) \
+                { \
+                    std::cout << typeid(T).name() << " " << #var << " does not match : " << double(lhs) << " - " << double(rhs) << "\n"; \
+                    abort(); \
```

Comment:
in general don't abort after a failed test if you can continue going, it will make it super hard to fix CI bugs (re-run pipeline to progress 1 bug at a time)